tests/AtlasSearchTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace MongoDB\Laravel\Tests;
+
+use Illuminate\Support\Facades\Schema;
+use MongoDB\Collection as MongoDBCollection;
+use MongoDB\Driver\Exception\ServerException;
+use MongoDB\Laravel\Schema\Builder;
+use MongoDB\Laravel\Tests\Models\Book;
+
+use function assert;
+use function usleep;
+
+class AtlasSearchTest extends TestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        Book::insert([
+            ['title' => 'Introduction to Algorithms'],
+            ['title' => 'Clean Code: A Handbook of Agile Software Craftsmanship'],
+            ['title' => 'Design Patterns: Elements of Reusable Object-Oriented Software'],
+            ['title' => 'The Pragmatic Programmer: Your Journey to Mastery'],
+            ['title' => 'Artificial Intelligence: A Modern Approach'],
+            ['title' => 'Structure and Interpretation of Computer Programs'],
+            ['title' => 'Code Complete: A Practical Handbook of Software Construction'],
+            ['title' => 'The Art of Computer Programming'],
+            ['title' => 'Computer Networks'],
+            ['title' => 'Operating System Concepts'],
+            ['title' => 'Database System Concepts'],
+            ['title' => 'Compilers: Principles, Techniques, and Tools'],
+            ['title' => 'Introduction to the Theory of Computation'],
+            ['title' => 'Modern Operating Systems'],
+            ['title' => 'Computer Organization and Design'],
+            ['title' => 'The Mythical Man-Month: Essays on Software Engineering'],
+            ['title' => 'Algorithms'],
+            ['title' => 'Understanding Machine Learning: From Theory to Algorithms'],
+            ['title' => 'Deep Learning'],
+            ['title' => 'Pattern Recognition and Machine Learning'],
+        ]);
+
+        $collection = $this->getConnection('mongodb')->getCollection('books');
+        assert($collection instanceof MongoDBCollection);
+        try {
+            $collection->createSearchIndex([
```

Comment:
An index with dynamic mappings should probably be included so you can assert its output.